### PR TITLE
SKW-4167: post transaction methods and cli

### PR DESCRIPTION
**Pull request by konvulse**
This change adds post transaction methods to the Python API.

Updates documentation and tests.

Fixes bad code in json_to_samples.

Not changing the version number until all API methods have been updated. (Let me know if that's a bad idea.)

### CODE REVIEW

**Comment by konvulse** [comment]
retest this please

**Comment by konvulse** [comment]
Only Conduce_Python_API_Freestyle is a real test.

**Comment by konvulse** [comment]
If this gets approved, please let me merge.  It conflicts with the code style PR.

**Comment by konvulse** [comment]
This PR has been rebased against charlie/pycodestyle

File: conduce/api.py [hunk 1]
```diff
@@ -598,6 +598,95 @@ def ingest_entities(dataset_id, entity_list, **kwargs):
     return _ingest_entity_set(dataset_id, entity_set, **kwargs)
 
 
+def insert_transaction(dataset_id, entity_set, **kwargs):
```
Comment (konvulse) [inline conduce/api.py]: It just occurred to me that these names may be too vague for the python API.  Should `insert_transaction` become `insert_dataset_transaction` and so on?

Comment (mbrennek) [reply conduce/api.py]: I think `insert_transaction` is fine.  We don't use transaction in any context other than around a dataset, correct?  What else might we add that has the concept of a transaction that must be inserted/appended/posted?

Comment (konvulse) [reply conduce/api.py]: No idea, terms like transaction and backend just don't seem specific (self-documenting) enough.

Comment (konvulse) [reply conduce/api.py]: For the backend APIs I used `create_dataset_backend`, `add_elasticsearch_store`, and `process_transactions`

**Pull request merged**